app/controllers/shopping_lists_controller.rb

```diff
@@ -18,6 +18,16 @@ def edit
     redirect_to @shopping_list
   end
 
+  def make_primary
+    ShoppingList.update_all(primary: false)
+
+    shopping_list = ShoppingList.find(params[:shopping_list_id])
```

Comment:
you can use a before_action for this if it's `ShoppingList.find(params[:shopping_list_id])` is being used elsewhere in the controller

Reply:
So far that's the one use of that finder. The other cases are `ShoppingList.find(params[:id])`.

I will keep this in mind! :)